catalog/plugin/catalog-plugin-metacard-validation/src/main/java/ddf/catalog/metacard/validation/MetacardValidityFilterPlugin.java

```diff
@@ -104,17 +108,20 @@ public PolicyResponse processPostQuery(Result input, Map<String, Serializable> p
         Metacard metacard = input.getMetacard();
         HashMap<String, Set<String>> securityMap = new HashMap<>();
 
-        if ((metacard.getAttribute(Validation.VALIDATION_ERRORS) != null && metacard.getAttribute(
-                Validation.VALIDATION_ERRORS)
-                .getValues() != null) || (
-                metacard.getAttribute(Validation.VALIDATION_WARNINGS) != null &&
-                        metacard.getAttribute(Validation.VALIDATION_WARNINGS)
-                                .getValues() != null)) {
+        if (filterErrors && metacard.getAttribute(Validation.VALIDATION_ERRORS) != null &&
+                metacard.getAttribute(Validation.VALIDATION_ERRORS)
+                        .getValues() != null) {
+            for (Map.Entry<String, List<String>> attributeMapping : attributeMap.entrySet()) {
```

Comment:
@coyotesqrl they do the same thing. The important part was the conditions of the `if` and `else if` so that depending on whether it was configured to filter errors, warnings, or both, either one of these can run (filter either or both) or neither of them (filter none).